test/unit/Cover/helpers.js

```diff
@@ -56,29 +56,33 @@ async function assertCoverFields(
   expect(segment.priceRatio).to.be.equal(targetPriceRatio);
 }
 
-async function buyCoverOnOnePool({
-  productId,
-  coverAsset,
-  period,
-  amount,
-  targetPriceRatio,
-  priceDenominator,
-  activeCover,
-  capacity,
-}) {
+async function buyCoverOnOnePool(params) {
+  return buyCoverOnMultiplePools.call(this, params, 1);
+}
+
+async function buyCoverOnMultiplePools(
```

Comment:
this is not clear enough what it does.

It both create pools and buys.

Also it's used in only 1 place.

Let's have a function that's buyCoverOnMultiplePools

but creates the pools before separately. There's only 1 place where this is used.

Reply:
I'm not sure I fully understood your proposed changes. I applied the following changes https://github.com/NexusMutual/smart-contracts/pull/462/commits/132e0886bf10f7ec6c5c7b5931b8959ccea48811 to only buy the cover on `buyCoverOnMultiplePools` and creating the pools outside the function, in the test that uses it and in `buyCoverOnOnePool` to not modify the behavior of that function as it is used in several unit tests.

Is something like this what you had in mind?

Reply:
This is what i mean yes, good!